### ISSUE TITLE
Team invites cannot be denied

### ISSUE THREAD

**Issue by venashial** [modrinth/labrinth#279]
**Describe the bug**
When invited to a team, the route to deny the invite (remove yourself from the team) doesn't work.

**To Reproduce**
1. Get invited to a team.
2. Remove yourself from the team
    `DELETE` `https://staging-api.modrinth.com/v2/team/Q6twuE5t/members/Y018kGLw`
3. See error
    `401 Unauthorized`
    ```json
    {
      "error": "unauthorized",
      "description": "Authentication Error: You don't have permission to edit members of this team"
    }
    ```

**Additional context**
The relevant code: https://github.com/modrinth/labrinth/blob/master/src/routes/teams.rs#L435-L441

**Comment by venashial** [comment]
The error isn't too apparent in the frontend because the notification is deleted first, and then it errors when trying to delete the user from the team. The team owner still sees the invite as pending, and cannot re-issue it until they delete the user and re-invite them.

**Comment by wafflecoffee** [comment]
Pretty sure this was fixed in #299, if not feel free to reopen
